packages/core/src/external/carequality/ihe-gateway-v2/ihe-gateway-v2-logic.ts

```diff
@@ -78,17 +147,13 @@ export async function createSignSendProcessDQRequests({
     bulkBodyData: dqRequestsGatewayV2,
     samlCertsAndKeys,
   });
-  const responses = await sendSignedDQRequests({
-    signedRequests,
-    samlCertsAndKeys,
-    patientId,
-    cxId,
-  });
-  const results = responses.map(response => {
-    return processDQResponse({
-      dqResponse: response,
-    });
+
+  const resultPromises = signedRequests.map(async (signedRequest, index) => {
+    return sendProcessRetryDqRequest({ signedRequest, samlCertsAndKeys, patientId, cxId, index });
   });
+
+  const results = await Promise.allSettled(resultPromises);
```

Comment:
When we use `allSettled` we need to parse the results to get successful and failed results.

Reply:
I meant to use promise.all then. Good catch

Reply:
Reviewing the logic of this function, upstream and downstream, it seems like `allSettled` would be better, no?

When we use `Promise.all()`, when the first promise throws/rejects the `.all()` throws that error. It means we don't know what happened with the the other promises on the array, and can't get the response from them.

`.allSettled()` solves this, but we need to process it to get the successful promises and the ones that failed.

It makes sense in this scenario, as we want o send the error to the OSS api, right?

Reply:
updated to use Promise.Allsettled

Reply:
Sorry, ended up moving to [another thread](https://github.com/metriport/metriport/pull/2268#discussion_r1643653977).